src/compile.rs

```diff
@@ -101,11 +105,70 @@ fn compile_universal2(
     Ok(result)
 }
 
+/// We want to use `zig cc` as linker and c compiler. We want to call `python -m ziglang cc`, but
+/// cargo only accepts a path to an executable as linker, so we add a wrapper script. We then also
+/// use the wrapper script to pass arguments and substitute an unsupported argument.
+///
+/// We create different files for different args because otherwise cargo might skip recompiling even
+/// if the linker target changed
+fn prepare_zig_linker(context: &BuildContext) -> Result<PathBuf> {
+    let (zig_linker, target) = match context.platform_tag {
+        None | Some(PlatformTag::Linux) => (
+            "./zigcc-gnu.sh".to_string(),
+            "native-native-gnu".to_string(),
+        ),
+        Some(PlatformTag::Musllinux { x, y }) => (
+            format!("./zigcc-musl-{}-{}.sh", x, y),
+            format!("native-native-musl.{}.{}", x, y),
+        ),
+        Some(PlatformTag::Manylinux { x, y }) => (
+            format!("./zigcc-gnu-{}-{}.sh", x, y),
+            format!("native-native-gnu.{}.{}", x, y),
+        ),
+    };
+
+    let zig_linker_dir = dirs::cache_dir()
+        // If the really is no cache dir, cwd will also do
+        .unwrap_or_else(|| PathBuf::from("."))
+        .join(env!("CARGO_PKG_NAME"))
+        .join(env!("CARGO_PKG_VERSION"));
+    fs::create_dir_all(&zig_linker_dir)?;
+    let zig_linker = zig_linker_dir.join(zig_linker);
+
+    #[cfg(target_family = "unix")]
+    let mut custom_linker_file = OpenOptions::new()
+        .create(true)
+        .write(true)
+        .truncate(true)
+        .mode(0o700)
+        .open(&zig_linker)?;
+    #[cfg(not(target_family = "unix"))]
+    let mut custom_linker_file = File::create(&zig_linker)?;
+    // https://github.com/ziglang/zig/issues/10050#issuecomment-956204098
+    custom_linker_file.write_all(
+        format!(
+            r##"#!/bin/bash
+python -m ziglang cc ${{@/-lgcc_s/-lunwind}} -target {}
```

Comment:
For musl I'm not whether we should replace libgcc_s with libunwind, for example, Alpine Linux has a patch to use libgcc_eh instead of libunwind:

https://github.com/alpinelinux/aports/blob/396a57da9bff572d15f17f57ff3f80b931d6e008/community/rust/0006-Prefer-libgcc_eh-over-libunwind-for-musl.patch

Reply:
I've made this to get rid of the manylinux docker and only filled out musllinux because I was there anyway, I've no problem with just disabling musl+zig, either by ignoring `--zig` on musl or by erroring on the combination

Reply:
I've tested musllinux with libgcc_s replaced with libunwind and it works fine, so I guess it's safe to do that. And the good thing is that it doesn't require repairing the wheel now for a pure Rust project.